src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -295,7 +295,7 @@ static bool TryParseXaml(SourceText text, string uid, Compilation compilation, A
 
 			if (rootClass != null)
 				XmlnsHelper.ParseXmlns(rootClass.Value, out rootType, out rootClrNamespace, out var rootAsm, out var targetPlatform);
-			else if (hasXamlCompilationProcessingInstruction)
+			else if (hasXamlCompilationProcessingInstruction && root.NamespaceURI == XamlParser.MauiUri)
```

Comment:
added this check to make sure the file was valid Xaml, but it ain't right, even if it works